zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/handlers/ListViewProcessInstanceFromIncidentHandlerTest.java

```diff
@@ -0,0 +1,175 @@
+        factory.generateRecord(ValueType.INCIDENT, r -> r.withValue(incidentValue));
+    return incidentRecord;
+  }
+
+  @Test
+  public void shouldCreateNewEntity() {
+    final var result = underTest.createNewEntity("id");
+    assertThat(result).isNotNull();
+    assertThat(result.getId()).isEqualTo("id");
+  }
+
+  @Test
+  public void shouldUpsertEntityOnFlush() {
+    // given
+    final ProcessInstanceForListViewEntity inputEntity =
+        new ProcessInstanceForListViewEntity()
+            .setId("111")
+            .setTreePath("PI_111/FN_3/FNI_456677/PI_23432");
+
+    final BatchRequest mockRequest = mock(BatchRequest.class);
+
+    final Map<String, Object> expectedUpdateFields = new LinkedHashMap<>();
+    expectedUpdateFields.put(TREE_PATH, inputEntity.getTreePath());
+    // when
+    underTest.flush(inputEntity, mockRequest);
+    // then
+    verify(mockRequest, times(1))
+        .upsert(indexName, inputEntity.getId(), inputEntity, expectedUpdateFields);
+  }
+
+  @Test
+  public void shouldUpdateEntityFromRecord() {
+    // given
+    final Long pi1Key = 111L;
+    final Integer callElementId1 = 3;
+    final long callActivity1Key = 123L;
+    final Long pi2Key = 222L;
+    final Integer callElementId2 = 6;
+    final long callActivity2Key = 234L;
+    final Long pi3Key = 333L;
+    final long flowNodeInstanceKey = 345L;
+    final Record<IncidentRecordValue> incidentRecord =
+        createIncidentRecord(
+            List.of(
+                List.of(pi1Key, callActivity1Key),
+                List.of(pi2Key, callActivity2Key),
+                List.of(pi3Key, flowNodeInstanceKey)),
+            List.of(callElementId1, callElementId2));
+    // when parent proces instance
+    final ProcessInstanceForListViewEntity processInstanceForListViewEntity1 =
+        new ProcessInstanceForListViewEntity().setId(String.valueOf(pi1Key));
+    underTest.updateEntity(incidentRecord, processInstanceForListViewEntity1);
+    // then
+    assertThat(processInstanceForListViewEntity1.getTreePath()).isEqualTo("PI_111");
+
+    // when called process 2nd level
+    final ProcessInstanceForListViewEntity processInstanceForListViewEntity2 =
+        new ProcessInstanceForListViewEntity().setId(String.valueOf(pi2Key));
+    underTest.updateEntity(incidentRecord, processInstanceForListViewEntity2);
+    // then
+    assertThat(processInstanceForListViewEntity2.getTreePath())
+        .isEqualTo("PI_111/FN_3/FNI_123/PI_222");
+    underTest.updateEntity(incidentRecord, processInstanceForListViewEntity2);
+
+    // when called process 3rd level
+    final ProcessInstanceForListViewEntity processInstanceForListViewEntity3 =
+        new ProcessInstanceForListViewEntity().setId(String.valueOf(pi3Key));
+    underTest.updateEntity(incidentRecord, processInstanceForListViewEntity3);
+    // then
+    assertThat(processInstanceForListViewEntity3.getTreePath())
+        .isEqualTo("PI_111/FN_3/FNI_123/PI_222/FN_6/FNI_234/PI_333");
```

Comment:
🤔 didn't we wanted to somehow extract the call activity element id via these index from the bpmn?

Reply:
it's done in another PR. 
https://github.com/camunda/camunda/pull/24511